src/classes/VOL_CTRL_PersonalSiteContactInfo_TEST.cls

```diff
@@ -30,61 +30,56 @@
 
 @isTest
 private with sharing class VOL_CTRL_PersonalSiteContactInfo_TEST {
-
+    private static VOL_Access_TEST.Stub accessMock = new VOL_Access_TEST.Stub();
+    
     //******************************************************************************************************
     // Test Code
 
+    private static void setAccessMock() {
+        VOL_CTRL_PersonalSiteContactInfo.access = (VOL_Access) Test.createStub(VOL_Access.class, accessMock);
+    }    
+    
+    private static testMethod void shouldCheckAccessOnSave(){
+        setAccessMock();
+		User guestUser = Util_UnitTest.guestUser;
+        Util_UnitTest.elevateGuestUserAccess();
+
+        Map<Id, SObject> sObjectById = Util_UnitTest.generateDataWithRecurrenceSchedules();
+
+        Contact contactRecord = [SELECT Id FROM Contact LIMIT 1];
+        System.AssertNotEquals(null, contactRecord);
+
+        // setup page
+        PageReference pageRef = Page.PersonalSiteContactInfo;
+        pageRef.getParameters().put('contactId', contactRecord.Id);
+        system.assertNotEquals(null, contactRecord.Id);   
+        Test.setCurrentPage(pageRef); 
+
+        VOL_CTRL_PersonalSiteContactInfo ctrl = new VOL_CTRL_PersonalSiteContactInfo();
+        ctrl.contactEdit.LastName = 'ChangedLastName';
+        ctrl.save();
+        
+        accessMock.assertMethodCalled('checkUpdateAccess', 1);
+        accessMock.assertMethodCalledWith('checkUpdateAccess', Contact.SObjectType);
+    }
+    
     /*******************************************************************************************************
     * @description test the visualforce page controller, running as the Sites Guest User, if such as user
     * exists.  if not, will run under the current user.
     * @return void
     ********************************************************************************************************/
-    private static testMethod void testPageWithSitesGuestUser() {
-        list<User> listU = [Select Username, UserType, Name, IsActive, Id From User where IsActive = true and UserType = 'Guest' and 
-            Name like '%Site Guest User%'];
-        if (listU.size() > 0) {
-            UTIL_UnitTest.elevateGuestUserAccess();
-            system.debug('Running test as Sites Guest User: ' + listU[0]);
-            system.runAs(listU[0]) {
-                testPSC();
-            }
-        } else {
-            system.debug('Running test as system.');
-            testPSC();
-        }
+    private static testMethod void testPageFunctionality() {
+        Map<Id, SObject> sObjectById = Util_UnitTest.generateDataWithRecurrenceSchedules();
```

Comment:
We could probably just call the method here without storing the results since it doesn't look like we use them.